### PR TITLE
feat(ui-common): refactor and update ImportDialog

### PR DESCRIPTION
**Pull request by skamril**
ANT-1713

### CODE REVIEW

File: webapp/src/components/common/FileTable.tsx [hunk 1]
```diff
@@ -238,8 +215,8 @@ function FileTable(props: PropType) {
       {openImportDialog && (
         <ImportDialog
           open={openImportDialog}
-          onClose={() => setOpenImportDialog(false)}
-          onImport={onImport}
+          onCancel={() => setOpenImportDialog(false)}
+          onImport={async (file) => uploadFile?.(file)}
```
Comment (hdinia) [inline webapp/src/components/common/FileTable.tsx:219]: optional: `onImport={(file) => uploadFile?.(file) ?? Promise.resolve()}`

File: webapp/src/components/common/dialogs/ImportDialog.tsx [hunk 2]
```diff
@@ -44,31 +62,35 @@ function ImportDialog(props: Props) {
...
-      } catch {
-        // noop
-      } finally {
-        if (isMounted()) {
-          setIsUploading(false);
-          setUploadProgress(-1);
-        }
-      }
+  async function handleDropAccepted(acceptedFiles: File[]) {
+    setInvalidText("");
```
Comment (hdinia) [inline webapp/src/components/common/dialogs/ImportDialog.tsx:66]: if used to display error messages maybe `setError`

File: webapp/src/components/common/dialogs/ImportDialog.tsx [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
   dropzoneText?: string;
   accept?: Accept;
-  onClose: VoidFunction;
+  onCancel: VoidFunction;
   onImport: (
     file: File,
-    onUploadProgress: (progress: number) => void,
-  ) => Promise<void>;
+    setUploadProgress: (progress: number) => void,
+  ) => PromiseAny;
```
Comment (hdinia) [inline webapp/src/components/common/dialogs/ImportDialog.tsx:19]: any reason to force any here ? I think in most cases its a void Promise or undefined

**Review by hdinia — APPROVED** [review]

File: webapp/src/components/common/FileTable.tsx (hunk 1 above)
Comment (skamril) [reply webapp/src/components/common/FileTable.tsx:219]: It's the same with one keyword

File: webapp/src/components/common/dialogs/ImportDialog.tsx (hunk 2 above)
Comment (skamril) [reply webapp/src/components/common/dialogs/ImportDialog.tsx:66]: I want to make a distinction between submit error (server side) and invalid value (front side) (like for field validation)

File: webapp/src/components/common/dialogs/ImportDialog.tsx (hunk 3 above)
Comment (skamril) [reply webapp/src/components/common/dialogs/ImportDialog.tsx:19]: The server call can return a value (like `importFile` in 'study.ts')

**Pull request merged**